scripts/skills/perks/perk_rf_ghostlike.nut

```diff
@@ -13,24 +20,73 @@ this.perk_rf_ghostlike <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
+	function isHidden()
+	{
+		return this.m.Enemies.len() == 0 && !this.getContainer().getActor().getCurrentProperties().IsImmuneToZoneOfControl;
+	}
+
 	function getTooltip()
 	{
-		local tooltip = this.skill.getTooltip();
+		local ret = this.skill.getTooltip();
+
+		if (this.getContainer().getActor().getCurrentProperties().IsImmuneToZoneOfControl)
+		{
+			ret.push({
+				id = 10,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = ::Reforged.Mod.Tooltips.parseString("The next movement will ignore [Zone of Control|Concept.ZoneOfControl]")
+			});
+		}
 
-		tooltip.push({
-			id = 6,
-			type = "text",
-			icon = "ui/icons/special.png",
-			text = ::Reforged.Mod.Tooltips.parseString("The next movement will ignore [Zone of Control|Concept.ZoneOfControl]")
-		});
+		if (this.m.Enemies.len() != 0)
+		{
+			ret.push({
+				id = 11,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = ::MSU.Text.colorizeMult(this.m.DamageTotalMult) + " more damage and " + ::MSU.Text.colorizeFraction(this.m.DirectDamageModifier, {AddSign = true}) + " damage ignoring armor against:"
+			});
 
-		return tooltip;
+			foreach (enemy in this.m.Enemies)
+			{
+				local enemy = ::Tactical.getEntityByID(enemy);
+				if (enemy == null || !enemy.isPlacedOnMap() || !enemy.isAlive())
+					continue;
+
+				ret.push({
+					id = 12,
+					type = "text",
+					icon = "ui/orientation/" + enemy.getOverlayImage() + ".png",
+					text = enemy.getName()
+				});
+			}
+
+			ret.push({
+				id = 13,
+				type = "text",
+				icon = "ui/icons/warning.png",
+				text = ::MSU.Text.colorRed("The damage bonus will be lost upon swapping an item or waiting or ending your turn")
+			});
+		}
+
+		return ret;
 	}
 
-	function onUpdate( _properties )
+	function isEnabled()
 	{
-		this.m.IsHidden = true;
+		if (this.m.IsForceEnabled)
```

Comment:
You don't check whether the weapon equipped is a melee weapon. There should be a check for that in front of this condition because IsForceEnabled does not turn that off.

Currently Ranged weapons (because they have reach 0) will also work with this perk

Reply:
In the new commit I've changed the design a bit to make the perk simpler and removed `IsForceEnabled`. It now requires up to 20 armor weight for the ignore zoc part, and requires melee weapon with <=4 reach for the damage part.